src/groupByKey.ts

```diff
@@ -1,7 +1,26 @@
+/* eslint-disable no-prototype-builtins */
 type GroupsMap<T> = {
   [key: string]: T[];
```

Comment:
to make it even better, you can make it so that the key can only be one of the keys of T.
```suggestion
  [key in keyof T]?: T[];
```

Reply:
your function already takes care of that, just to double-check it here